### PR TITLE
feat: Implement paid drop

### LINKED ISSUE #8835: Implement paid drop
### Is your feature request related to a problem?

Drop API has new feature called `'price'` so you can check price per mint of NFT

### Describe the solution you would like

1. Show price per mint on /[prefix]/drops
2. Shows price on button when minting
3. implement price minting - it is similar to holder of (the minting `api.tx.nfts.mint`)
except

```ts
    const mint = api.tx.nfts.mint(
      collectionId.value,
      collectionRes.items,
      accountId.value,
      {
        ownedItem: null,
        mintPrice: price,
      })
```

### Describe alternatives you have considered

rest stays the same

May @Jarsen136 can pick it up. 

Please also check for other issues .

For testing I made /ahk/drops/coretime :) 

### Screenshots

_No response_

### Are there opened related issues?

_No response_

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

  👇 __ Let's make a quick check before the contribution.

  ## PR Type

  - [x] Feature

  ## Needs QA check

  - @kodadot/qa-guild please review

  ## Context

  - [x] Closes #8835
  - [ ] Requires deployment <snek/rubick/worker>

  #### Did your issue had any of the "$" label on it?

  - [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)

  ## Screenshot 📸

  - [ ] My fix has changed UI

  ## Copilot Summary
  copilot:summary

  copilot:poem

### CODE REVIEW

**Comment by yangwao** [comment]
release incoming 👀

**Comment by vikiival** [comment]
@Jarsen136 enabled

File: components/collection/drop/PaidGenerative.vue [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+      {
+        ownedItem: null,
+        mintPrice: price.value,
+      },
+    )
+
+    const transfer = api.tx.balances.transfer(
+      '5GGWQ1yiSvS2rPciRtAuK2xQTuxCcgoGZ7dTSzHWws4ELzwD',
+      2e9,
+    )
```
Comment (vikiival) [inline components/collection/drop/PaidGenerative.vue:203]: remove, this is already part on line 196

File: components/collection/drop/PaidGenerative.vue [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+    const { apiInstance } = useApi()
+    const api = await apiInstance.value
+    const collectionRes = (
+      await api.query.nfts.collection(collectionId.value)
+    ).toJSON() as {
+      items: string
+    }
+
+    initTransactionLoader()
+    const cb = api.tx.utility.batchAll
```
Comment (vikiival) [inline components/collection/drop/PaidGenerative.vue:189]: no need for utility batch

File: pages/[prefix]/drops/[id].vue [hunk 3]
```diff
@@ -4,6 +4,12 @@
   <CollectionDropHolderOfGenerative
     v-else-if="dropType === 'generative' && drop.alias === 'chroma'"
     :drop="drop" />
+  <CollectionDropPaidGenerative
+    v-else-if="
+      dropType === 'generative' &&
+      PAID_GENERATIVE_ALIAS_LIST.includes(drop.alias)
```
Comment (vikiival) [inline pages/[prefix]/drops/[id].vue:10]: You can just check it it has price

**Review by vikiival — CHANGES_REQUESTED** [review]
Smol

**Comment by Jarsen136** [comment]
There is a CORS issue related with the vercel screenshot service. Could you check it? @vikiival 

related to https://github.com/kodadot/workers/issues/244
image

File: components/collection/drop/PaidGenerative.vue (hunk 2 above)
Comment (Jarsen136) [reply components/collection/drop/PaidGenerative.vue]: done

File: pages/[prefix]/drops/[id].vue (hunk 3 above)
Comment (Jarsen136) [reply pages/[prefix]/drops/[id].vue]: done

**Comment by Jarsen136** [comment]
✅ tested `/ahk/gallery/330-1`

image

image

**Comment by yangwao** [comment]
LGTM

**Comment by vikiival** [comment]
@Jarsen136 coretime is active too, no need to mock

**Comment by Jarsen136** [comment]
> @Jarsen136 coretime is active too, no need to mock

No worry, I only mocked it on my localhost and made a screenshot.

**Comment by prury** [comment]
can we set date accordingly? 
image

**Comment by Jarsen136** [comment]
> can we set date accordingly? image
> 
> proper mint phase since is not yet open image


✅ Both of them were fixed

**Comment by prury** [comment]
@Jarsen136 could you also fix artist twitter? its leading to https://twitter.com/pifragile_ instead of /pifragile

**Comment by Jarsen136** [comment]
> @Jarsen136 could you also fix artist twitter? its leading to https://twitter.com/pifragile_ instead of /pifragile

It's intended and it's from the description of the collection
https://canary.kodadot.xyz/ahp/collection/66
image

**Comment by vikiival** [comment]
Coming from @JustLuuuu 

Will fix


 image

**Comment by vikiival** [comment]
image

@prury its correct on canary

**Comment by prury** [comment]
> @prury its correct on canary

huh?

**Comment by yangwao** [comment]
I guess it's 7pm UTC, 5pm UTC+2

**Comment by exezbcz** [comment]
we are utc, so it should be 5pm UTC no?

**Comment by vikiival** [comment]
I commited date fix (from mobile 🫣) should be up on canary soon

**Comment by hassnian** [comment]
@Jarsen136  image screenshot failing. is that normal ? I'm using coretime drop

**Comment by vikiival** [comment]
Should not be failing

**Comment by Jarsen136** [comment]
> There is a CORS issue related with the vercel screenshot service. Could you check it? @vikiival
> 
> related to [kodadot/workers#244](https://github.com/kodadot/workers/issues/244) image

Did you encounter this CORS issue on localhost? @hassnian

**Comment by vikiival** [comment]
@hassnian whats is the output of the fail?

White screen or same pic as collection

**Comment by hassnian** [comment]
yep cors

CleanShot 2024-01-04 at 14 44 37@2x


CleanShot 2024-01-04 at 14 44 30@2x

**Comment by hassnian** [comment]
> @hassnian whats is the output of the fail?
> 
> White screen or same pic as collection

same pic

**Comment by hassnian** [comment]
@vikiival can /claim return nftName and Image ?

**Comment by vikiival** [comment]
CORS was fixed, works bit differently than on CFW 




Screenshot 2024-01-04 at 11 45 32

**Comment by vikiival** [comment]
Screenshot 2024-01-04 at 12 03 21


so some requests are dead in the que. They should spawn in matter of minutes (I disabled kusama last time)

**Comment by vikiival** [comment]
> @vikiival can /claim return nftName and Image ?

Name is returned, 
image can be returned too if it is sent to `/claim` request cc @Jarsen136

**Comment by Jarsen136** [comment]
> > @vikiival can /claim return nftName and Image ?
> 
> Name is returned, image can be returned too if it is sent to `/claim` request cc @Jarsen136

I have added the `image` parameter for the `/claim` request.

**Comment by Jarsen136** [comment]
How about merging this PR now? Then we could go to check the following-up issue #8846 @vikiival @yangwao

**Comment by vikiival** [comment]
LGTM to me, 


Note for other:
except the stuff with who minted (but it is on my side.)

**Review by vikiival — APPROVED** [review]

**Pull request merged**